src/main/java/at/ac/tuwien/kr/alpha/Main.java

```diff
@@ -27,256 +27,76 @@
-import org.slf4j.LoggerFactory;
-
-import java.io.FileInputStream;
 import java.io.FileNotFoundException;
 import java.io.IOException;
-import java.nio.charset.CodingErrorAction;
-import java.nio.file.Paths;
-import java.util.Arrays;
-import java.util.HashSet;
-import java.util.Random;
-import java.util.Set;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-import static at.ac.tuwien.kr.alpha.Util.literate;
-import static at.ac.tuwien.kr.alpha.Util.streamToChannel;
-import static java.nio.file.Files.lines;
+import org.antlr.v4.runtime.RecognitionException;
+import org.apache.commons.cli.ParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import at.ac.tuwien.kr.alpha.common.AnswerSet;
+import at.ac.tuwien.kr.alpha.common.Program;
+import at.ac.tuwien.kr.alpha.config.AlphaConfig;
+import at.ac.tuwien.kr.alpha.config.CommandLineParser;
+import at.ac.tuwien.kr.alpha.config.InputConfig;
+import at.ac.tuwien.kr.alpha.solver.Solver;
+import at.ac.tuwien.kr.alpha.solver.SolverMaintainingStatistics;
 
 /**
  * Main entry point for Alpha.
  */
 public class Main {
-	private static final Logger LOGGER = LoggerFactory.getLogger(Main.class);
-	private static final ProgramParser PARSER = new ProgramParser();
-
-	private static final String OPT_INPUT = "input";
-	private static final String OPT_HELP  = "help";
-	private static final String OPT_NUM_AS  = "numAS";
-	private static final String OPT_GROUNDER = "grounder";
-	private static final String OPT_SOLVER = "solver";
-	private static final String OPT_FILTER = "filter";
-	private static final String OPT_STRING = "str";
-	private static final String OPT_SORT = "sort";
-	private static final String OPT_DETERMINISTIC = "deterministic";
-	private static final String OPT_STORE = "store";
-	private static final String OPT_QUIET = "quiet";
-	private static final String OPT_LITERATE = "literate";
-	private static final String OPT_STATS = "stats";
-	private static final String OPT_NO_JUSTIFICATION = "disableJustifications";
-	private static final String OPT_NORMALIZATION_GRID = "normalizationCountingGrid";
-
-	private static final String OPT_BRANCHING_HEURISTIC = "branchingHeuristic";
-	private static final String DEFAULT_GROUNDER = "naive";
-	private static final String DEFAULT_SOLVER = "default";
-	private static final String DEFAULT_STORE = "alphaRoaming";
-	private static final String OPT_SEED = "seed";
-	private static final String OPT_DEBUG_INTERNAL_CHECKS = "DebugEnableInternalChecks";
-	private static final String DEFAULT_BRANCHING_HEURISTIC = Heuristic.NAIVE.name();
 
-	private static final java.util.function.Predicate<Predicate> DEFAULT_FILTER = p -> true;
+	private static final Logger LOGGER = LoggerFactory.getLogger(Main.class);
 
-	private static CommandLine commandLine;
+	private static final String ALPHA_CALL_SYNTAX = "java -jar alpha-bundled.jar" + System.lineSeparator() + "java -jar alpha.jar";
```

Comment:
Isn´t it the case that the artefacts are usually called `Alpha-bundled.jar` and `Alpha.jar`? I think this is because the Github project name starts with an upper-case A.

Reply:
Depends on the name of the Gradle project inside of which the task to produce the bundled JAR is run. On my machine the project source code, i.e. this repository is placed inside a directory called "alpha" and not "Alpha" for personal (and therefore, subjective) convenience. This results in lowercase JAR names.

Reply:
I realize that the directory name can be changed, that´s why I used the word "usually". No magic string that we are using here will always be right, but isn´t it the case that if you clone the repo from GitHub it will by default have an upper-case A?

Reply:
We could also try to use the [actual name of the JAR file](https://stackoverflow.com/questions/14189162/get-name-of-running-jar-or-exe).

Reply:
I did a quick test and the "actual name of jar file" variant works if I do it like this:
`String appname = new File(Main.class.getProtectionDomain().getCodeSource().getLocation().toURI()).getName();
		CommandLineParser commandLineParser = new CommandLineParser(appname, (msg) -> Main.exitWithMessage(msg, 0));`

That works fine both when I do a "java -jar" on the bundled jar and when I call the Main class directly with "java -cp Alpha-bundled.jar at.ac.tuwien.kr.alpha.Main".
**However**, you don't get any useful results from that if you don't have the class in a jar file when calling it from command line (I tested this using the eclipse "execute button" since I didn't want to type out the complete classpath).
Furthermore,  "...getLocation().toUri()" throws a URISyntaxException if whatever code location we use isn't representable as a URI - I'm not sure how likely that happening is, but I'm kinda reluctant to deal with the potential case for something that's really just a "nice to have".

The cleanest way I see to do this would be:
- adapt gradle config  to add the build artifact name into the META-INF/MANIFEST.MF file of the jar and read it from there
- if there is no manifest file (explicit check, no exception handler), fall back to "<your-alpha-jar>.jar"

However, even this doesn't address the (admittedly super-picky) point that we wouldn't be able to automatically test the format of the usage message with a unit test since the output would be influenced by how the test runner loaded the class.

Please let me know your thoughts!

Reply:
To me this sounds fine. Thanks!